From 9cf3a2ccfa342a27a6863ea998ec5e57ece909ed Mon Sep 17 00:00:00 2001
From: Dang Huynh <danct12@riseup.net>
Date: Sun, 3 Dec 2023 00:46:41 +0700
Subject: [PATCH 07/12] arm64: dts: qcom: sm6115-pro1x: Enable MDSS and GPU

Fxtec Pro1x uses the same display (BOE BF060Y8M-AJ0) as Pro1.

Signed-off-by: Dang Huynh <danct12@riseup.net>
---
 .../boot/dts/qcom/sm6115-fxtec-pro1x.dts      | 60 +++++++++++++++++--
 1 file changed, 56 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sm6115-fxtec-pro1x.dts b/arch/arm64/boot/dts/qcom/sm6115-fxtec-pro1x.dts
index 7ac698fac9bf..025315059d68 100644
--- a/arch/arm64/boot/dts/qcom/sm6115-fxtec-pro1x.dts
+++ b/arch/arm64/boot/dts/qcom/sm6115-fxtec-pro1x.dts
@@ -93,13 +93,16 @@ ts_vddio_supply: ts-vddio-supply {
 	};
 };
 
-&dispcc {
-	/* HACK: disable until a panel driver is ready to retain simplefb */
-	status = "disabled";
+&gpi_dma0 {
+	status = "okay";
 };
 
-&gpi_dma0 {
+&gpu {
 	status = "okay";
+
+	zap-shader {
+		firmware-name = "qcom/sm6115/Fxtec/QX1050/a610_zap.mbn";
+	};
 };
 
 &i2c1 {
@@ -136,6 +139,42 @@ touchscreen@14 {
 	};
 };
 
+&mdss {
+	status = "okay";
+};
+
+&mdss_dsi0 {
+	vdda-supply = <&pm6125_l18a>;
+	status = "okay";
+
+	panel: panel@0 {
+		compatible = "boe,bf060y8m-aj0";
+		reg = <0>;
+
+		reset-gpios = <&tlmm 82 GPIO_ACTIVE_LOW>;
+
+		vddio-supply = <&pm6125_l9a>;
+
+		pinctrl-0 = <&mdss_dsi_n &panel_en_n>;
+		pinctrl-names = "default";
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&mdss_dsi0_out>;
+			};
+		};
+	};
+};
+
+&mdss_dsi0_out {
+	data-lanes = <0 1 2 3>;
+	remote-endpoint = <&panel_in>;
+};
+
+&mdss_dsi0_phy {
+	status = "okay";
+};
+
 &pm6125_gpios {
 	vol_up_n: vol-up-n-state {
 		pins = "gpio5";
@@ -345,6 +384,19 @@ key_camera_n: key-camera-n {
 		bias-pull-up;
 	};
 
+	mdss_dsi_n: mdss-dsi-n {
+		pins = "gpio82";
+		function = "gpio";
+		drive-strength = <8>;
+		bias-disable;
+	};
+
+	panel_en_n: panel-en-n {
+		pins = "gpio65";
+		function = "gpio";
+		bias-disable;
+	};
+
 	ts_int_n: ts-int-n {
 		pins = "gpio80";
 		function = "gpio";
-- 
2.45.2

